Content.Server/SurveillanceCamera/Systems/SurveillanceCameraMapSystem.cs

```diff
@@ -0,0 +1,88 @@
+using System.Numerics;
+using Content.Server.Power.Components;
+using Content.Shared.DeviceNetwork.Components;
+using Content.Shared.SurveillanceCamera.Components;
+
+namespace Content.Server.SurveillanceCamera;
+
+public sealed class SurveillanceCameraMapSystem : EntitySystem
+{
+    [Dependency] private readonly SharedTransformSystem _transform = default!;
+
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<SurveillanceCameraComponent, ComponentStartup>(OnCameraStartup);
+        SubscribeLocalEvent<SurveillanceCameraComponent, MoveEvent>(OnCameraMoved);
+
+        SubscribeNetworkEvent<RequestCameraMarkerUpdateMessage>(OnRequestCameraMarkerUpdate);
+    }
+    private void OnCameraStartup(EntityUid uid, SurveillanceCameraComponent comp, ComponentStartup args)
+    {
+        UpdateCameraMarker(uid, comp);
+    }
+
+    private void OnCameraMoved(EntityUid uid, SurveillanceCameraComponent comp, ref MoveEvent args)
+    {
+        var oldGridUid = args.OldPosition.GetGridUid(EntityManager);
+        var newGridUid = args.NewPosition.GetGridUid(EntityManager);
+
+        if (oldGridUid != newGridUid && oldGridUid is not null)
+        {
+            if (TryComp<SurveillanceCameraMapComponent>(oldGridUid, out var oldMapComp))
+            {
+                var netEntity = GetNetEntity(uid);
+                if (oldMapComp.Cameras.Remove(netEntity))
+                    Dirty(oldGridUid.Value, oldMapComp);
+            }
+        }
+
+        if (newGridUid is not null)
+            UpdateCameraMarker(uid, comp);
+    }
+
+    private void OnRequestCameraMarkerUpdate(RequestCameraMarkerUpdateMessage args)
+    {
+        var cameraEntity = GetEntity(args.CameraEntity);
+
+        if (TryComp<SurveillanceCameraComponent>(cameraEntity, out var comp) && HasComp<DeviceNetworkComponent>(cameraEntity))
+            UpdateCameraMarker(cameraEntity, comp);
+    }
+
+    public void UpdateCameraMarker(EntityUid uid, SurveillanceCameraComponent comp)
```

Comment:
In general and especially for public APIs it's preferred to use the `Entity<T>` style argument signature. In this case you could do `UpdateCameraMarker(Entity<SurveillanceCameraComponent> ent)`. That could then also include the `DeviceNetworkComponent` and `TransformComponent` which you can then get via `Resolve`. See also https://hackmd.io/@Slart/S1hsoGFm1l#Pairing-uids-and-Components.

This also should get docs since it's public.